lib/smack/SmackInstGenerator.cpp

```diff
@@ -343,17 +362,35 @@ void SmackInstGenerator::visitStoreInst(llvm::StoreInst& si) {
   processInstruction(si);
   const llvm::Value* P = si.getPointerOperand();
   const llvm::Value* E = si.getOperand(0);
-  const Expr* rhs = rep.expr(E);
-  const llvm::GlobalVariable* G = llvm::dyn_cast<const llvm::GlobalVariable>(P);
 
-  if (rep.isFloat(E))
-    rhs = Expr::fn("$fp2si", rhs);
+  if (rep.tryBitVector()) {
```

Comment:
No need for such complicated branching, there are only two cases. You can simplify:
````
if (!rep.tryBitVector() || rep.tryDSA() && rep.isFieldDisjoint(...)) {
  emit(Stmt::assign(...));
} else {
  emit(rep.loadAsBytes(...));
}No need for such complicated branching, there are only two cases. You can simplify:
````
if (!rep.tryBitVector() || rep.tryDSA() && rep.isFieldDisjoint(...)) {
  emit(Stmt::assign(...));
} else {
  emit(rep.storeAsBytes(...));
}